builtin/logical/aws/path_roles.go

```diff
@@ -442,7 +460,29 @@ func (r *awsRoleEntry) toResponseData() map[string]interface{} {
 		"policy_document": r.PolicyDocument,
 		"default_sts_ttl": int64(r.DefaultSTSTTL.Seconds()),
 		"max_sts_ttl":     int64(r.MaxSTSTTL.Seconds()),
+		"user_path":       r.UserPath,
+	}
+
+	// Remove fields from response if it is not valid to provide them
+	// in write operation with specific credential_type
```

Comment:
So you are saying that other backends also doesn't hold consistency between write input and read output? If that's so and you think it is the best to leave this behaviour I don't know how to fix issues related to #5972. Tools like terraform-vault-provider check if there are any changes between configuration and reality by matching input and output. But again if you think it causes more problems than it solves I will definitely revert these changes

Reply:
I think there are two different workflows. One is "specify the entirety of the configuration every time." The other is "specify the things that you want to change and don't touch things that aren't specified." The former is great if you're using a tool like Terraform to manage your Vault settings and just do a `terraform apply` for every settings change. But it kinda sucks if you're not using something like Terraform. As an extreme example, it can be problematic when you're updating configs like `/config/client` -- that endpoint can take in secrets like an AWS secret access key and non-secrets like the STS endpoint to use. Should a Vault operator really be required to know the AWS secret access key in order to change the STS endpoint to a different region?

Vault auth backends generally (as far as I am aware, certainly the AWS auth backend does) take the latter approach but enable former. If you don't specify a config value, then it doesn't get touched. If you want to un-set a config value, then you have to explicitly pass it in as an empty string (in most cases). Then, the onus is on declarative tools like Terraform to detect if extra config options have been set than are being requested and, if so, pass those in as empty strings to get Vault to un-set them.

Reply:
Thank for such extensive explanation. But I still don't understand why return fields which you cannot set when providing specific credential_type? Lets say i provide `iam_user` - aws backend will not even write fields like `default_sts_ttl` or `max_sts_ttl`to path but just adds them in response. I dont see reason why user needs that information in output and plus it does not allow use terraform properly. So by removing these fields from output I dont see how it breaks these workflows :)

Reply:
Does it really cause a problem with Terraform? In my experience, if you don't set a value, Terraform will just treat it as the golang zero value, and so I wouldn't expect it to cause problems. The lack of an input value in the Terraform should tell Terraform to expect a zero value back from the resource.

Reply:
Thanks for the answer again :) The problem is similar to #5972. Because terraform compares what you set and what you get (write input with read output) in order to detect changes. With current setup terraform will always complain that there is drift because we can't match write input with read output :) I am not sure if I explained very clearly, let me know :)

Reply:
Hi @povils ! 

Those are excellent points!

I may be mistaken, but I think Terraform provides a variety of ways to deal with input and output not being the same. The [schema](https://github.com/hashicorp/terraform/blob/master/helper/schema/schema.go#L61) for each field, for instance, allows one to supply a [DiffSuppressFunc](https://github.com/hashicorp/terraform/blob/master/helper/schema/schema.go#L142) if needed. 

I really like keeping Vault's design concerns separate from Terraform's wherever possible. Do you think it would be reasonable to deal with diff issues in the [Terraform Vault Provider's](https://github.com/terraform-providers/terraform-provider-vault) code rather than by altering Vault's API?

Reply:
Thanks for response. I think it makes sense just to separate these concerns :)